play-json-jsoniter/src/main/scala/com/evolutiongaming/jsonitertool/PlayJsonJsoniter.scala

```diff
@@ -5,99 +5,14 @@ import java.io.{InputStream, OutputStream}
-        } else if (b == 'f' || b == 't') {
-          in.rollbackToken()
-          if (in.readBoolean()) JsTrue else JsFalse
-        } else if ((b >= '0' && b <= '9') || b == '-') {
-          in.rollbackToken()
-          val s = JsonParserSettings.settings.bigDecimalParseSettings
-          JsNumber(in.readBigDecimal(null, s.mathContext, s.scaleLimit, s.digitsLimit))
-        } else if (b == '[') {
-          val array: IndexedSeq[JsValue] =
-            if (in.isNextToken(']')) new Array[JsValue](0)
-            else {
-              in.rollbackToken()
-              var i = 0
-              var arr = new Array[JsValue](4)
-              do {
-                if (i == arr.length) arr = java.util.Arrays.copyOf(arr, i << 1)
-                arr(i) = decodeValue(in, default)
-                i += 1
-              } while (in.isNextToken(','))
-
-              if (in.isCurrentToken(']'))
-                if (i == arr.length) arr else java.util.Arrays.copyOf(arr, i)
-              else in.arrayEndOrCommaError()
-            }
-          JsArray(array)
-        } else if (b == '{') {
-           /*
-            * Because of DoS vulnerability in Scala 2.12 HashMap https://github.com/scala/bug/issues/11203
-            * we use a Java LinkedHashMap because it better handles hash code collisions for Comparable keys.
-            */
-          val kvs =
-            if (in.isNextToken('}')) new java.util.LinkedHashMap[String, JsValue]()
-            else {
-              val underlying = new java.util.LinkedHashMap[String, JsValue]()
-              in.rollbackToken()
-              do {
-                underlying.put(in.readKeyAsString(), decodeValue(in, default))
-              } while (in.isNextToken(','))
-
-              if (!in.isCurrentToken('}'))
-                in.objectEndOrCommaError()
-
-              underlying
-            }
-          import scala.jdk.CollectionConverters._
-          JsObject(kvs.asScala)
-        } else in.decodeError("expected JSON value")
-      }
-
-      override def encodeValue(jsValue: JsValue, out: JsonWriter): Unit =
-        jsValue match {
-          case JsBoolean(b) =>
-            out.writeVal(b)
-          case JsString(value) =>
-            out.writeVal(value)
-          case JsNumber(value) =>
-            out.writeVal(value)
-          case JsArray(items) =>
-            out.writeArrayStart()
-            items.foreach(encodeValue(_, out))
-            out.writeArrayEnd()
-          case JsObject(kvs) =>
-            out.writeObjectStart()
-            kvs.foreach {
-              case (k, v) =>
-                out.writeKey(k)
-                encodeValue(v, out)
-            }
-            out.writeObjectEnd()
-          case JsNull =>
-            out.writeNull()
-        }
-
-      override val nullValue: JsValue = JsNull
-    }
+  implicit val jsValueCodec: JsonValueCodec[JsValue] = {
+    val settings = JsonParserSettings.settings
+    JsonValueCodecJsValue(settings.bigDecimalParseSettings)
```

Comment:
Maybe JsonParserSettings type should be used as a constructor argument as in has 2 subset of settings (bigDecimalSerializerSettings and bigDecimalParseSettings)

Reply:
Why if it other fields are kit used